### PR TITLE
Add dobi-powered Docker build pipelines

### PR DESCRIPTION
**Pull request by nkierpiec**
Chef Expeditor now uses [Dobi](https://dnephin.github.io/dobi/index.html) to power its Docker image builds using a specialized Buildkite pipeline. You can find out more about these pipelines in our [Expeditor Documentation](https://expeditor.chef.io/docs/pipelines/docker/). 

## Description

- Added dobi.yaml with specs for how to build
- Added build.docker.yml shim pipeline definition.
- Updated relevant files that use expeditor/dobi

Signed-off-by: Nathaniel Kierpiec <nkierpiec@chef.io>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: .envrc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export IMAGE_REGISTRY="localhost:5000"
```
Comment (tas50) [inline .envrc:1]: Having this file here is going to clobber potential valid settings for anyone working in this repo using direnv

File: .expeditor/build.docker.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+image_registry: chef
+env:
+  VERSION: "16.3.18"
```
Comment (tas50) [inline .expeditor/build.docker.yml:3]: is this something we can just source from the VERSION file so we can maintain that as the source of truth for buildkite/expeditor actions

Comment (nkierpiec) [reply .expeditor/build.docker.yml:3]: I added that to the update-version.sh file already so it should be handled that way

File: .envrc (hunk 1 above)
Comment (nkierpiec) [reply .envrc:1]: How do you propose we should go about handling this then? Maybe just adding to the README as a prereq to what is needed to run dobi?

File: README.md [hunk 3]
```diff
@@ -44,6 +44,10 @@ To learn more about our monthly feature releases and yearly major releases, see
 
 We'd love to have your help developing Chef Infra. See our [Contributing Document](./CONTRIBUTING.md) for more information on getting started.
 
+## How to use [Dobi](https://dnephin.github.io/dobi/)
```
Comment (tas50) [inline README.md:47]: This should probably just be a Building Docker Container doc here: https://github.com/chef/chef/tree/master/docs/dev

**Pull request merged**